dbms/src/Flash/Pipeline/Schedule/Events/tests/gtest_event.cpp

```diff
@@ -499,7 +500,8 @@ try
     {
         PipelineExecutorStatus exec_status;
         auto event = std::make_shared<ManyTasksEvent>(exec_status, i);
-        event->schedule();
+        if (event->prepreForSource())
+            event->schedule();
```

Comment:
Before this pr, all the event will be scheduled without check the dependency?

Reply:
Before this pr, only event can be schedule immediately `event.withoutInput` == true.
But it doesn't matter if `event.withoutInput` doesn't call.
In this pr, `event.prepareForSource` must be called before `event.schedule`